public_html/wp-content/plugins/pattern-directory/includes/favorite.php

```diff
@@ -84,3 +84,81 @@ function get_favorites( $user = 0 ) {
 
 	return array_map( 'absint', $favorites );
 }
+
+/**
+ * Get the cached count of how many times this pattern has been favorited.
+ *
+ * @param int|WP_Post $post The pattern to check.
+ * @return integer
+ */
+function get_favorite_count( $post = 0 ) {
+	$post = get_block_pattern( $post );
+	if ( ! $post ) {
+		return false;
+	}
+
+	return absint( get_post_meta( $post->ID, META_KEY, true ) );
+}
+
+/**
+ * Get a count of how many times this pattern has been favorited, directly from the users table.
+ *
+ * @param int|WP_Post $post The pattern to check.
+ * @return integer
+ */
+function get_raw_favorite_count( $post = 0 ) {
+	global $wpdb;
+	$post = get_block_pattern( $post );
+	if ( ! $post ) {
+		return false;
+	}
+	$count = $wpdb->get_var( $wpdb->prepare(
+		"SELECT COUNT(*)
+			FROM {$wpdb->usermeta}
+			WHERE meta_key=%s
+			AND meta_value=%d",
+		META_KEY,
+		$post->ID
+	) );
+
+	return absint( $count );
+}
+
+/**
+ * Update a given post's favorite count cache.
+ *
+ * @param mixed $value The post ID.
+ */
+function update_favorite_cache( $value ) {
+	if ( ! is_numeric( $value ) ) {
+		return;
+	}
+
+	$count = get_raw_favorite_count( $value );
+	update_post_meta( $value, META_KEY, $count );
```

Comment:
```suggestion
	$count = get_raw_favorite_count( $value );
	
	if ( ! is_int( $count ) ) {
		return;
	}

	update_post_meta( $value, META_KEY, $count );
```

I think this is a bit cleaner since `get_raw_favorite_count` is really where you check that you have a valid pattern post.

Reply:
Thanks, I think `update_favorite_cache` is more understandable now 👍🏻